### PR TITLE
cmake: delete SYSTEM_CLANG and auto-download mechanism

### PR DESCRIPTION
**Pull request by MaskRay**
Notably, delete `SYSTEM_CLANG` which downloads prebuilt clang+llvm archives from releases.llvm.org

### CODE REVIEW

**Comment by MaskRay** [comment]
This does not work currently. I must have missed something:

```
% cmake -H. -Bbuild -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_COMPILER=~/llvm/Stable/bin/clang++ -DCMAKE_EXPORT_COMPILE_COMMANDS=On -DCMAKE_PREFIX_PATH="$LLVM/Release;$LLVM/Release/tools/clang;$LLVM;$LLVM/tools/clang"
-- Could NOT find Clang (missing: Clang_LIBRARY Clang_INCLUDE_DIR LLVM_INCLUDE_DIR LLVM_BUILD_INCLUDE_DIR) (found suitable version "8.0.0", minimum required is "7.0.0")
-- Configuring done
CMake Error at CMakeLists.txt:12 (add_executable):
  Target "ccls" links to target "Clang::Clang" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?


CMake Error at CMakeLists.txt:12 (add_executable):
  Target "ccls" links to target "Clang::Clang" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
```

@DaanDeMeyer I'm sorry to bother you but would you like to help?

**Comment by DaanDeMeyer** [comment]
Your changes work for me (Arch Linux, clang+llvm packages installed). Most likely your CMAKE_PREFIX_PATH isn't entirely correct.

File: CMakeLists.txt [hunk 1]
```diff
@@ -143,7 +106,7 @@ target_include_directories(ccls SYSTEM PRIVATE
 
 install(TARGETS ccls RUNTIME DESTINATION bin)
 
-if(NOT SYSTEM_CLANG AND NOT ${CMAKE_SYSTEM_NAME} STREQUAL Windows)
+if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL Windows)
```
Comment (DaanDeMeyer) [inline CMakeLists.txt]: I'm pretty sure this entire if can be removed

File: CMakeLists.txt [hunk 2]
```diff
@@ -159,7 +122,7 @@ if(NOT SYSTEM_CLANG AND NOT ${CMAKE_SYSTEM_NAME} STREQUAL Windows)
 endif()
 
 # Allow running from build Windows by copying libclang.dll to build directory
-if(NOT SYSTEM_CLANG AND ${CMAKE_SYSTEM_NAME} STREQUAL Windows)
```
Comment (DaanDeMeyer) [inline CMakeLists.txt]: Same here

File: CMakeLists.txt (hunk 1 above)
Comment (MaskRay) [reply CMakeLists.txt]: Thanks. Removed

File: CMakeLists.txt (hunk 2 above)
Comment (MaskRay) [reply CMakeLists.txt]: Deleted

**Pull request merged**